internal/cmd/parse_quotas.go

```diff
@@ -0,0 +1,151 @@
+//   service/resource=123(Unit)
+// where unit is optional.
+func parseToQuotaRequest(resUnits resourceUnits, in []string) (limes.QuotaRequest, error) {
+	out := make(limes.QuotaRequest)
+	for _, inputStr := range in {
+		matchList := quotaRx.FindStringSubmatch(inputStr)
+		if matchList == nil {
+			return nil, fmt.Errorf("expected a quota/capacity with optional unit and comment in the format: service/resource=value(unit):comment, got %q", inputStr)
+		}
+
+		// Validate input service/resource.
+		service := matchList[1]
+		resource := matchList[2]
+		defaultUnit, ok := resUnits[service][resource]
+		if !ok {
+			return nil, fmt.Errorf("invalid resource: %s/%s does not exist in Limes", service, resource)
+		}
+
+		valStr := matchList[3]
+		isFloatVal := strings.Contains(valStr, ".")
+		unitStr := matchList[4]
+
+		// Validate input unit.
+		var unit limes.Unit
+		if unitStr == "" {
+			if isFloatVal {
+				return nil, fmt.Errorf("counted (i.e. resource without any unit) values must be an integer, got %q", valStr)
+			}
+			unit = limes.UnitNone
+		} else {
+			for _, v := range core.LimesUnits {
+				if unitStr == string(v) {
+					unit = v
+					break
+				}
+			}
+			if unit == "" {
+				return nil, fmt.Errorf("expected a unit from %q, got %q", core.LimesUnits, unitStr)
+			}
+		}
+
+		// Validate and convert (if needed) input value.
+		var vWithUnit limes.ValueWithUnit
+		if isFloatVal {
+			logg.Info("Limes only accepts integer values, will attempt to convert %s %s to a suitable unit for %s/%s)",
+				valStr, unit, service, resource)
+			var err error
+			vWithUnit, err = convertTo(valStr, unit, defaultUnit)
+			if err != nil {
+				if _, ok := err.(limes.IncompatibleUnitsError); ok {
+					err = fmt.Errorf("can not convert %s %s, minimum accepted unit for %s/%s is %s",
+						valStr, unit, service, resource, defaultUnit)
+				}
+				return nil, err
+			}
+		} else {
+			v, err := strconv.ParseUint(valStr, 10, 64)
+			if err != nil {
+				return nil, err
+			}
+			vWithUnit = limes.ValueWithUnit{Value: v, Unit: unit}
+		}
+
+		if _, ok := out[service]; !ok {
+			out[service] = limes.ServiceQuotaRequest{
+				Resources: make(limes.ResourceQuotaRequest),
+			}
+		}
+		out[service].Resources[resource] = vWithUnit
+	}
+
+	return out, nil
+}
+
+// convertTo converts the given value from source to target unit and returns
+// the truncated result in limes.ValueWithUnit.
+// i.e: 22.65 TiB -> 23193 GiB (instead of 23193.6 GiB)
+func convertTo(valStr string, source, target limes.Unit) (limes.ValueWithUnit, error) {
+	v, err := strconv.ParseFloat(valStr, 64)
+	if err != nil {
+		return limes.ValueWithUnit{}, err
+	}
+
+	var newV float64
+	if source == target {
+		newV = math.Floor(v)
+	} else {
+		_, sourceMultiple := source.Base()
+		_, targetMultiple := target.Base()
+		if sourceMultiple < targetMultiple {
+			return limes.ValueWithUnit{}, limes.IncompatibleUnitsError{Source: source, Target: target}
+		}
```

Comment:
This should also check if the base units match. Right now all measured units are based on bytes, but that could change in the future.